scripts/minify.js

```diff
@@ -21,9 +25,33 @@ const minifyOptions = {
     }
 }
 
-async function minifyFile(file){
+async function minifyFile(file, options){
+    options = options || {};
+    let newOptions = _.defaultsDeep({...options}, minifyOptions)
 
-    const minifiedResult = await minify(file, minifyOptions);
+    if(/\.js/.test(file)){
+        //special case for JS, since the lib does not generate map files (even not return the map data)
+        //minfy and generate map file locally
+        const data = await readFile(file, 'utf8');
+
+        const { error, code, map } = terser.minify(data, options.js);
+        
+        if (error)
+            throw error;
+
+        if(options.js.sourceMap){
+            try {
+                await createDir(options.js.sourceMap.filename)
+                await writeFile(options.js.sourceMap.filename, map)
```

Comment:
maybe better to specify `utf8` encoding
https://nodejs.org/api/fs.html#fs_fs_writefile_file_data_options_callback
```suggestion
                await writeFile(options.js.sourceMap.filename, map, 'utf8')
```